### ISSUE TITLE
setting scaleway_ip reverse DNS to the IP of the server is impossible

### ISSUE THREAD

**Issue by ahanselka** [terraform-providers/terraform-provider-scaleway#88]
This isn't necessarily a bug per se, but if you want to set your reverse DNS of the `scaleway_ip` to the name/hostname of the server in question while also managing DNS records with terraform for the same server, it will fail because it requires the server and DNS records to be created first. The Scaleway API won't let you set a reverse DNS without a matching forward DNS, so it is impossible to workaround by using variables instead of the data variables.

The below configuration returns a Cycle since the `scaleway_ip` and the `scaleway_server` would then depend on each other. I'm not well versed on `terraform` internals, but would it be possible to re-visit the reverse DNS after the server and DNS records are created? Alternatively, are there any suggestions on how to achieve what I wish to achieve?

Additionally, the `scaleway_ip` resource will try to unset the reverse DNS if it is not set within terraform, so managing it without the reverse DNS would also be problematic I think.

### Terraform Version

Terraform v0.11.7
+ provider.google v1.8.0
+ provider.scaleway v1.5.1

### Affected Resource(s)

scaleway_ip
scaleway_server

### Terraform Configuration Files
```hcl
variable "node_name" {}
variable "image" {}
variable "type" {}

variable "ipv6" {
  default = "true"
}

data "google_dns_managed_zone" "dns-zone" {
  name = "dns-zone-name"
}

resource "scaleway_ip" "server-ip" {
  reverse = "${scaleway_server.generic-scw.node_name}.${data.google_dns_managed_zone.dns-zone.dns_name}."
}

resource "scaleway_server" "generic-scw" {
  name                = "${var.node_name}"
  image               = "${var.image}"
  type                = "${var.type}"
  enable_ipv6         = "${var.ipv6}"
  public_ip           = "${scaleway_ip.server-ip.ip}"
}

resource "google_dns_record_set" "external" {
  name         = "${var.node_name}.${data.google_dns_managed_zone.dns-zone.dns_name}"
  type         = "A"
  ttl          = 3600
  managed_zone = "${data.google_dns_managed_zone.dns-zone.name}"
  rrdatas      = ["${scaleway_server.generic-scw.public_ip}"]
}
```

**Comment by nicolai86** [comment]
Thank you @ahanselka for reporting this. I’ll investigate if we can make this work :)

**Comment by ahanselka** [comment]
I was thinking on this last night and one solution might be to decouple reverse DNS from `scaleway_ip`, perhaps by adding a `scaleway_ip_reverse` or similar resource that could then thus `depends_on` the DNS record.

**Comment by nicolai86** [comment]
@ahanselka why are you not using the `server` attribute on `scaleway_ip` ?

```
variable "node_name" {}
variable "image" {}
variable "type" {}

variable "ipv6" {
  default = "true"
}

data "google_dns_managed_zone" "dns-zone" {
  name = "dns-zone-name"
}

resource "scaleway_ip" "server-ip" {
  reverse = "${scaleway_server.generic-scw.node_name}.${data.google_dns_managed_zone.dns-zone.dns_name}."
  server = "${scaleway_server.generic-scw.id}"
}

resource "scaleway_server" "generic-scw" {
  name                = "${var.node_name}"
  image               = "${var.image}"
  type                = "${var.type}"
  enable_ipv6         = "${var.ipv6}"
}

resource "google_dns_record_set" "external" {
  name         = "${var.node_name}.${data.google_dns_managed_zone.dns-zone.dns_name}"
  type         = "A"
  ttl          = 3600
  managed_zone = "${data.google_dns_managed_zone.dns-zone.name}"
  rrdatas      = ["${scaleway_server.generic-scw.public_ip}"]
}
```

that should fix your issue and work as expected.

**Comment by nicolai86** [comment]
Fixed with the new reverse dns resource in 1.8
